File: C#/AwesomeShop/AwesomeShop.Sales/Events/OnboardingCustomerCreatedSubscriber.cs
using Microsoft.AspNetCore.Connections;
using RabbitMQ.Client;
using RabbitMQ.Client.Events;
using System.Text;

namespace AwesomeShop.Sales.Events
{
    public class OnboardingCustomerCreatedSubscriber : BackgroundService
    {
        private readonly IConnection _connection;
        private readonly IModel _channel;
        private const string Queue = "sales-service/customer-created";
        public OnboardingCustomerCreatedSubscriber()
        {
            var connectionFactory = new ConnectionFactory
            {
                HostName = "localhost"
            };

            _connection = connectionFactory.CreateConnection("sales-service-customer-created-consumer");

            _channel = _connection.CreateModel();
        }
        protected override Task ExecuteAsync(CancellationToken stoppingToken)
        {
            var consumer = new EventingBasicConsumer(_channel);

            consumer.Received += async (sender, eventArgs) =>
            {
                var contentArray = eventArgs.Body.ToArray();
                var message = Encoding.UTF8.GetString(contentArray);

                Console.WriteLine($"Message CustomerCreatedEvent received with {message}");

                _channel.BasicAck(eventArgs.DeliveryTag, false);
            };

            _channel.BasicConsume(Queue, false, consumer);

            return Task.CompletedTask;
        }
    }
}

File: VB/AwesomeShop/AwesomeShop.Sales/Events/OnboardingCustomerCreatedSubscriber.vb
Imports RabbitMQ.Client
Imports RabbitMQ.Client.Events
Imports System.Text
Imports System.Threading
Imports Microsoft.Extensions.Hosting

Public Class OnboardingCustomerCreatedSubscriber
    Inherits BackgroundService

    Private ReadOnly _connection As IConnection
    Private ReadOnly _channel As IModel
    Private Const Queue As String = "sales-service/customer-created"

    Public Sub New()
        Dim connectionFactory As New ConnectionFactory With {
            .HostName = "localhost"
        }

        _connection = connectionFactory.CreateConnection("sales-service-customer-created-consumer")
        _channel = _connection.CreateModel()
    End Sub

    Protected Overrides Function ExecuteAsync(ByVal stoppingToken As CancellationToken) As Task
        Dim consumer As New EventingBasicConsumer(_channel)

        AddHandler consumer.Received, Async Sub(sender, eventArgs)
                                          Dim contentArray As Byte() = eventArgs.Body.ToArray()
                                          Dim message As String = Encoding.UTF8.GetString(contentArray)

                                          Console.WriteLine($"Message CustomerCreatedEvent received with {message}")

                                          _channel.BasicAck(eventArgs.DeliveryTag, False)
                                      End Sub

        _channel.BasicConsume(Queue, False, consumer)

        Return Task.CompletedTask
    End Function
End Class
